desktop/libs/notebook/src/notebook/connectors/sql_alchemy.py

```diff
@@ -251,9 +252,25 @@ def execute(self, notebook, snippet):
         }
       )
 
-    result = connection.execute(statement)
+    async_ = self.interpreter['dialect'] == 'hive'
+    if async_:
+      result = connection.execute(statement, async_=async_)
+    else:
+      result = connection.execute(statement)
+
+    logs = []
+
+    if result.cursor and hasattr(result.cursor, 'poll'):
+      status = result.cursor.poll().operationState
+
+      if result.cursor and hasattr(result.cursor, 'fetch_logs'):
+        logs += result.cursor.fetch_logs()
+
+        while status in (TOperationState.INITIALIZED_STATE, TOperationState.RUNNING_STATE):
```

Comment:
Question -> I was running a query that never ends, so in this case also status is not changing i.e. code is not running till the completion of `result = connection.execute(statement, async_=async_)`  so when we have running status?

Reply:
Do we get a query id/handle after `connection.execute(statement, async_=True)` ?

Reply:
we have this only after the query completion

Reply:
What do you mean? Does it block or not?

https://github.com/dropbox/PyHive#db-api-asynchronous is for DBAPI, it might be different param for SqlAlchemy